src/cgutils.cpp

```diff
@@ -2722,12 +2722,14 @@ static jl_cgval_t emit_new_struct(jl_codectx_t &ctx, jl_value_t *ty, size_t narg
                                 ConstantInt::get(T_size, jl_field_offset(sty, i) + jl_field_size(sty, i) - 1))));
             }
         }
-        bool need_wb = false;
         // TODO: verify that nargs <= nf (currently handled by front-end)
         for (size_t i = 0; i < nargs; i++) {
             const jl_cgval_t &rhs = argv[i];
-            if (jl_field_isptr(sty, i) && !rhs.isboxed)
```

Comment:
What was the condition that required this previously? It was pre linear IR so I won't be surprised if it isn't needed anymore but I do remember it was quite non-obvious so I want to make sure it's not needed anymore...

Reply:
I thought it should be OK now because of linear IR, but I could be wrong. CI seems OK with it.